rocketchat-core/src/test/java/com/rocketchat/core/RestImplTest.java

```diff
@@ -88,51 +116,115 @@ public void tesSigninShouldBeSuccessfull() {
+    //    \_____|______|  |_|      |_|  \_\\____/ \____/|_|  |_|      |_|    |_____|______|______|_____/     |_|  |______|_____/   |_| |_____/
+    //
+    //
+
+    @Test(expected = NullPointerException.class)
+    public void testGetRoomFilesShouldFailWithNullRoomId() {
+        rest.getRoomFiles(null, BaseRoom.RoomType.PUBLIC, "0", Attachment.SortBy.UPLOADED_DATE, Sort.DESC, paginatedCallback);
+    }
+
     @Test(expected = NullPointerException.class)
-    public void testSigninShouldFailWithNullUsername() {
-        sut.signin(null, "password", loginCallback);
+    public void testGetRoomFilesShouldFailWithNullRoomType() {
+        rest.getRoomFiles("roomId", null, "0", Attachment.SortBy.UPLOADED_DATE, Sort.DESC, paginatedCallback);
     }
 
     @Test(expected = NullPointerException.class)
-    public void testSigninShouldFailWithNullPassword() {
-        sut.signin("username", null, loginCallback);
+    public void testGetRoomFilesShouldFailWithNullOffset() {
+        rest.getRoomFiles("roomId", BaseRoom.RoomType.PUBLIC, null, Attachment.SortBy.UPLOADED_DATE, Sort.DESC, paginatedCallback);
     }
 
     @Test(expected = NullPointerException.class)
-    public void testSigninShouldFailWithNullCallback() {
-        sut.signin("username", "password", null);
+    public void testGetRoomFilesShouldFailWithNullSortBy() {
+        rest.getRoomFiles("roomId", BaseRoom.RoomType.PUBLIC, "0", null, Sort.DESC, paginatedCallback);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testGetRoomFilesShouldFailWithNullSort() {
+        rest.getRoomFiles("roomId", BaseRoom.RoomType.PUBLIC, "0", Attachment.SortBy.UPLOADED_DATE, null, paginatedCallback);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void testGetRoomFilesShouldFailWithNullCallback() {
+        rest.getRoomFiles("roomId", BaseRoom.RoomType.PUBLIC, "0", Attachment.SortBy.UPLOADED_DATE, Sort.DESC, null);
+    }
+
+    @Test
+    public void testGetRoomFilesShouldFailOnInvalidJson() {
+        mockServer.expect()
+                .get()
+                .withPath("/api/v1/channels.files")
+                .andReturn(200, "NOT A JSON")
+                .once();
+
+        rest.getRoomFiles("general", BaseRoom.RoomType.PUBLIC, "0", Attachment.SortBy.UPLOADED_DATE, Sort.DESC, paginatedCallback);
+        verify(paginatedCallback, timeout(100).only())
+                .onError(exceptionCaptor.capture());
+
+        RocketChatException exception = exceptionCaptor.getValue();
+        assertThat(exception.getMessage(), is(equalTo("A JSONObject text must begin with '{' at character 0")));
+        assertThat(exception.getCause(), is(instanceOf(JSONException.class)));
+    }
+
+    @Test
+    //TODO Needs to know why it is failing.
+    public void testGetRoomFilesShouldBeSuccessful() {
+        rest.getRoomFiles("general", BaseRoom.RoomType.PUBLIC, 0, Attachment.SortBy.UPLOADED_DATE, Sort.DESC, paginatedCallback);
```

Comment:
You need  to configure the `mockServer` to expect a `files` call and return a response...

Reply:
Check `tesSigninShouldBeSuccessful()` for that